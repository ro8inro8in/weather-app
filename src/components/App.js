import React, { useState } from "react";
import "../styles/App.css";
import PropTypes from "prop-types";
import "../styles/ForecastSummaries.css";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";

const App = ({ forecasts, location }) => {
  const [selectedDate, setSelectedDate] = useState(forecasts[0].date);
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };
  return (
    <div className="weather-app">
      <LocationDetails city={location.city} country={location.country} />
      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
      />
      <ForecastDetails forecast={selectedForecast} />
    </div>
  );
};

App.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
        min: PropTypes.number,
      }),
    })
  ).isRequired,
  location: PropTypes.shape({
    city: PropTypes.string,
    country: PropTypes.string,
  }).isRequired,
};

export default App;
