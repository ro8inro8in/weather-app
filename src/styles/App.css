body {  
  margin: 16px;  
  font-size: 100%;  
  background-color: rgb(68, 126, 250);
  font-family: 'Open Sans', sans-serif;
}  
.weather-app {
  margin: 0 50px;
  display: flex;
  flex-direction: column;
  align-items: center;
}
