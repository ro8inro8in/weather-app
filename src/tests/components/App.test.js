import React from "react";
import { render } from "@testing-library/react";
import App from "../../components/App";

describe("App", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<App />);
    expect(asFragment).toMatchSnapshot();
  });
  // xit('renders button with correct text', () => {
  //   const { getByRole } = render(<App />);
  //   expect(getByRole("button")).toBeTruthy(); 
  // });
  // xit('calls correct function on click', () => {
  //   const onClick = jest.fn();
  //   const { getByText } = render(
  //     <App onClick={onClick} />
  //   );
  //   userEvent.click(getByText(defaultProps.text));
  //   expect(onClick).toHaveBeenCalled();
  // });
});