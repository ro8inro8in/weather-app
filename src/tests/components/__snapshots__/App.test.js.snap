// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `[Function]`;
